### ISSUE TITLE
Github CGNS version problem in parallel

### ISSUE THREAD

**Issue by mostafa-najafiyazdi** [CGNS/CGNS#5]
The Github version has problem when the CGNS is compiled for parallel and used in a parallel code, and when either of the following functions are called:

- cg_open
- cg_is_cgns

Note, that I made sure these files are called only on the master CPU. What happens is that the code easily runs when lunched with a single CPU, but with 2 or more CPUs, the master CPU gets glinched in these functions and never comes out. 

The library on SourceForge doesn't seem to have this problem.

**Comment by brtnfld** [comment]
If are not actually using cgns as a parallel reader/writer then you should not compile it with enable parallel. Enable parallel is for actually reading/writing cgns files in parallel, meaning all the processors process the CGNS file. If you are just having one processor read/write to the file then just compile CGNS without enable-parallel. 
The cg_open (actually cgp_open, which is just a wrapper for cg_open) is a collective operation, meaning all the processes have to call the it. This was changed from earlier versions because CGNS was getting terrible parallel read/write performance and cgp_open was a big problem.

**Comment by brtnfld** [comment]
Closing as duplicate of #414 which tracks the same issue with more detail.
